ls1intum/Artemis
# `Programming exercises`: Fix missing branch configuration

### Checklist
#### General




### Motivation and Context
The branch attribute was missing in the client. Therefore, whenever an instructor edits a programming exercise, the branch was set to null in the database which is problematic, because the branch information might be needed in follow-up operations.

### Description
Since the user should not even be able to change the branch name after creation, we override it in the update endpoint before saving the entity.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a new programming exercise
2. Open the edit screen: verify in the chrome console (or database) that the branch is available
3. Save the programming exercise and open the edit page again: verify in the chrome console (or database) that the branch is available

### Review Progress






#### Code Review
#### Manual Tests
